codeface/R/cluster/graph_comparison.r

```diff
@@ -142,37 +173,134 @@ graph.comparison <- function(g.1, g.2) {
   g.1.intersect <- graph.adjacency(adj.matrix.1.intersect, mode = "directed")
   g.2.intersect <- graph.adjacency(adj.matrix.2.intersect, mode = "directed")
 
-  graph.diff <- graph.difference(g.1.intersect, g.2.intersect)
+  graph.diff <- graph.difference(g.1.intersect, g.2.intersect, weighted = weighted, symmetric = symmetric)
 
   return(graph.diff)
 }
 
+get.index.map1 <- function(ids) {
+  N        <- nrow(ids)
+  map      <- new.env(size=N)
+  for(i in 1:N) {
+    map[[as.character(ids$id1[i])]] <- ids$id[i]
+  }
+  return(map)
+}
 
+get.index.map2 <- function(ids) {
+  N        <- nrow(ids)
+  map      <- new.env(size=N)
+  for(i in 1:N) {
+    map[[as.character(ids$id2[i])]] <- ids$id[i]
+  }
+  return(map)
+}
```

Comment:
@matthid Yeah R takes some getting used to, no worries. Perhaps you could pass in the column names as parameters instead of creating two functions. (see above)